ext/webserial/01_webserial.js

```diff
@@ -0,0 +1,397 @@
+    }
+
+    getInfo() {
+      webidl.assertBranded(this, SerialPort);
+      return this[_info];
+    }
+
+    // deno-lint-ignore require-await
+    async open(options) {
+      webidl.assertBranded(this, SerialPort);
+      const prefix = "Failed to execute 'open' on 'SerialPort'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
+      options = webidl.converters.SerialOptions(options, {
+        prefix,
+        context: "Argument 1",
+      });
+
+      if (this[_state] !== "closed") {
+        throw new DOMException(
+          "SerialPort must be closed",
+          "InvalidStateError",
+        );
+      }
+
+      if (options.dataBits !== 7 || options.dataBits !== 8) {
+        throw new TypeError("Invalid 'dataBits' given, must be either 7 or 8.");
+      }
+      if (options.stopBits !== 1 || options.stopBits !== 2) {
+        throw new TypeError("Invalid 'stopBits' given, must be either 1 or 2.");
+      }
+      if (options.bufferSize === 0) {
+        throw new TypeError(
+          "Invalid 'bufferSize' given, must be greater than 0.",
+        );
+      }
+
+      this[_state] = "opening";
+      this[_rid] = core.opSync("op_webserial_open_port", this[_name], options);
+      this[_state] = "opened";
+      this[_bufferSize] = options.bufferSize;
+    }
+
+    async setSignals(options = {}) {
+      webidl.assertBranded(this, SerialPort);
+      const prefix = "Failed to execute 'setSignals' on 'SerialPort'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
+      options = webidl.converters.SerialPortRequestOptions(options, {
+        prefix,
+        context: "Argument 1",
+      });
+
+      if (this[_state] !== "opened") {
+        throw new DOMException(
+          "SerialPort must be opened",
+          "InvalidStateError",
+        );
+      }
+
+      // TODO(@crowlKats): If all of the specified members of signals are not present reject promise with TypeError and return promise.
+      await core.opAsync("op_webserial_set_signals", this[_rid], options);
+    }
+
+    async getSignals() {
+      webidl.assertBranded(this, SerialPort);
+      if (this[_state] !== "opened") {
+        throw new DOMException(
+          "SerialPort must be opened",
+          "InvalidStateError",
+        );
+      }
+
+      return await core.opAsync("op_webserial_get_signals", this[_rid]);
+    }
+
+    async close() {
+      webidl.assertBranded(this, SerialPort);
+      let cancelPromise;
+      if (this[_readable] === null) {
+        cancelPromise = PromiseResolve(undefined);
+      } else {
+        cancelPromise = this[_readable].cancel();
+      }
+
+      let abortPromise;
+      if (this[_writable] === null) {
+        abortPromise = PromiseResolve(undefined);
+      } else {
+        abortPromise = this[_writable].abort();
+      }
+
+      const pendingClosePromise = new Promise((res) => {
+        if (this[_readable] === null && this[_writable] === null) {
+          res(undefined);
```

Comment:
I presume it is WebIDL spec conformity that these need to be resolved with an explicit undefined instead of zero-args?

Reply:
easier to follow and match with spec